src/vector_index.h

```diff
@@ -29,34 +29,50 @@
   QERR_MKBADARGS_FMT(status, "Missing mandatory parameter: cannot create %s index without specifying %s argument", algorithm, arg)
 
 typedef enum {
-  VECTOR_SIM_INVALID = 0,
-  VECTOR_SIM_TOPK = 1,
+  VECSIM_QT_TOPK,
 } VectorQueryType;
 
-typedef struct VectorFilter {
-  char *property;                 // name of field
-  void *vector;                   // vector data
-  size_t vecLen;                  // vector length
-  VectorQueryType type;           // TOPK
-  bool isBase64;                  // uses base64 strings
-  long long efRuntime;            // efRuntime
-  double value;                   // can hold int for TOPK or double for RANGE.
+typedef enum {
+  VECSIM_RUN_KNN = 0,
+  VECSIM_RUN_BATCH = 1,
+} VectorQueryRunType;
+
+typedef struct {
+  const char *name;
+  size_t namelen;
+  const char *value;
+  size_t vallen;
+  bool isParam;
+} VectorQueryParam;
```

Comment:
Add comments about this struct and it purpose. Why it will always hold `<string, string>`